main.R

```diff
@@ -0,0 +1,707 @@
+  geco_2022,
+  technology = ifelse(
+    .data$sector == "hdv", paste0(.data$technology, "_hdv"), .data$technology
+  )
+)
+
+geco_2022 <- dplyr::mutate(
+  geco_2022,
+  scenario_geography = dplyr::case_when(
+    .data$scenario_geography == "NOAP" ~ "Algeria & Libya",
+    .data$scenario_geography == "MEME" ~ "Mediteranean Middle East",
+    .data$scenario_geography == "NOAN" ~ "Morocco & Tunisia",
+    .data$scenario_geography == "NZL" ~ "New Zealand",
+    .data$scenario_geography == "RCIS" ~ "Other CIS",
+    .data$scenario_geography == "RCAM" ~ "Rest Central America",
+    .data$scenario_geography == "RCEU" ~ "Other Balkan",
+    .data$scenario_geography == "RSAM" ~ "Rest South America",
+    .data$scenario_geography == "RSAS" ~ "Rest South Asia",
+    .data$scenario_geography == "RSEA" ~ "Rest South East Asia",
+    .data$scenario_geography == "RSAF" ~ "Rest Sub Saharan Africa",
+    .data$scenario_geography == "RGLF" ~ "Rest Gulf",
+    .data$scenario_geography == "RPAC" ~ "Rest Pacific",
+    .data$scenario_geography == "KOR" ~ "South Korea",
+    .data$scenario_geography == "World" ~ "Global",
+    .data$scenario_geography == "THA" ~ "Thailand",
+    .data$scenario_geography == "EU" ~ "EU27",
+    .data$scenario_geography == "NOR" ~ "Norway",
+    .data$scenario_geography == "ISL" ~ "Iceland",
+    .data$scenario_geography == "CHE" ~ "Switzerland",
+    .data$scenario_geography == "TUR" ~ "Turkey",
+    .data$scenario_geography == "RUS" ~ "Russia",
+    .data$scenario_geography == "USA" ~ "US",
+    .data$scenario_geography == "CAN" ~ "Canada",
+    .data$scenario_geography == "BRA" ~ "Brazil",
+    .data$scenario_geography == "ARG" ~ "Argentina",
+    .data$scenario_geography == "CHL" ~ "Chile",
+    .data$scenario_geography == "AUS" ~ "Australia",
+    .data$scenario_geography == "JPN" ~ "Japan",
+    .data$scenario_geography == "CHN" ~ "China",
+    .data$scenario_geography == "IND" ~ "India",
+    .data$scenario_geography == "SAU" ~ "Saudi Arabia",
+    .data$scenario_geography == "IRN" ~ "Iran",
+    .data$scenario_geography == "EGY" ~ "Egypt",
+    .data$scenario_geography == "ZAF" ~ "South Africa",
+    .data$scenario_geography == "MEX" ~ "Mexico",
+    .data$scenario_geography == "IDN" ~ "Indonesia",
+    .data$scenario_geography == "UKR" ~ "Ukraine",
+    .data$scenario_geography == "MYS" ~ "Malaysia",
+    .data$scenario_geography == "VNM" ~ "Vietnam",
+    .data$scenario_geography == "GBR" ~ "UK",
+    TRUE ~ .data$scenario_geography
+  )
+)
+
+geco_2022 <- dplyr::ungroup(geco_2022)
+geco_2022 <- dplyr::filter(geco_2022, .data$source == "GECO2022")
+
+
+geco_2022 <- dplyr::mutate(
+  geco_2022,
+  scenario = dplyr::case_when(
+    grepl("1.5", .data$scenario) ~ "1.5C",
+    grepl("NDC", .data$scenario) ~ "NDC_LTS",
+    grepl("Ref", .data$scenario) ~ "Reference",
+    TRUE ~ NA_character_
+  )
+)
+
+if (any(is.na(unique(geco_2022$scenario)))) {
+  stop("Unique scenario names are not well-defined. Please review!")
```

Comment:
NB: consider making this more informative by extracting the scenarios that result in `NA`

```r
unknown_scenarios <- dplyr::filter(is.na(.data$scenario))
logger::log_error("Unknown scenario found: \"{unique(unknown_scenarios$scenario_rawname})\"")
```

This would require preserving the raw scenario name in the table, at least through this check, but you can also use it to give the user an idea of what to look for (rather than the current generic error)

Reply:
I will also leave this as is, since the error would only pick out a `NA` scenario name (which isn't that helpful for debugging)